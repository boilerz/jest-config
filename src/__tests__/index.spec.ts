import frontendConfig from '../frontend';
import config from '../index';
import reactNativeConfig from '../react-native';

describe('jest config', () => {
  it('should return default jest config', () => {
    expect(config).toMatchSnapshot();
  });

  it('should return frontend jest config', () => {
    expect(frontendConfig).toMatchSnapshot();
  });

  it('should return react-native jest config', () => {
    expect(reactNativeConfig).toMatchSnapshot();
  });
});
