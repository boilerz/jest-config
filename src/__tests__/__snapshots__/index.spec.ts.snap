// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest config should return default jest config 1`] = `
Object {
  "collectCoverageFrom": Array [
    "<rootDir>/src/**/*.{ts,tsx,js,jsx}",
  ],
  "moduleFileExtensions": Array [
    "js",
    "json",
    "jsx",
    "ts",
    "tsx",
    "node",
  ],
  "modulePathIgnorePatterns": Array [
    "<rootDir>/dist",
    "<rootDir>/lib",
    "<rootDir>/coverage",
  ],
  "resetMocks": true,
  "restoreMocks": true,
  "testEnvironment": "node",
  "transform": Object {
    "^.+\\\\.tsx?$": "ts-jest",
  },
}
`;

exports[`jest config should return frontend jest config 1`] = `
Object {
  "collectCoverageFrom": Array [
    "<rootDir>/src/**/*.{ts,tsx,js,jsx}",
  ],
  "moduleFileExtensions": Array [
    "js",
    "json",
    "jsx",
    "ts",
    "tsx",
    "node",
  ],
  "modulePathIgnorePatterns": Array [
    "<rootDir>/dist",
    "<rootDir>/lib",
    "<rootDir>/coverage",
  ],
  "resetMocks": true,
  "restoreMocks": true,
  "testEnvironment": "jsdom",
  "transform": Object {
    "^.+\\\\.tsx?$": "ts-jest",
  },
}
`;

exports[`jest config should return react-native jest config 1`] = `
Object {
  "collectCoverageFrom": Array [
    "<rootDir>/src/**/*.{ts,tsx,js,jsx}",
  ],
  "moduleFileExtensions": Array [
    "js",
    "json",
    "jsx",
    "ts",
    "tsx",
    "node",
  ],
  "modulePathIgnorePatterns": Array [
    "<rootDir>/dist",
    "<rootDir>/lib",
    "<rootDir>/coverage",
  ],
  "preset": "react-native",
  "resetMocks": true,
  "restoreMocks": true,
  "transform": Object {
    "^.+\\\\.tsx?$": "ts-jest",
  },
}
`;
